WeakAuras/GenericTrigger.lua

```diff
@@ -3679,10 +3679,26 @@ function WeakAuras.RegisterItemCountWatch()
   if not(itemCountWatchFrame) then
     itemCountWatchFrame = CreateFrame("Frame");
     itemCountWatchFrame:RegisterUnitEvent("UNIT_SPELLCAST_SUCCEEDED", "player");
-    itemCountWatchFrame:SetScript("OnEvent", function()
+    itemCountWatchFrame:RegisterEvent("BAG_UPDATE_COOLDOWN")
+    itemCountWatchFrame:RegisterEvent("BAG_UPDATE_DELAYED")
+    local batchUpdateCount = function()
+      itemCountWatchFrame:SetScript("OnUpdate", nil)
+      WeakAuras.ScanEvents("ITEM_COUNT_UPDATE")
+    end
+    itemCountWatchFrame:SetScript("OnEvent", function(self, event)
       Private.StartProfileSystem("generictrigger");
-      timer:ScheduleTimer(WeakAuras.ScanEvents, 0.2, "ITEM_COUNT_UPDATE");
-      timer:ScheduleTimer(WeakAuras.ScanEvents, 0.5, "ITEM_COUNT_UPDATE");
+      if event == "BAG_UPDATE_COOLDOWN" then
+        -- workaround to blizzard bug: refreshing healthstones from soulwell dont trigger BAG_UPDATE_DELAYED
+        -- so, we fake it by listening to B_U_C and checking on next frame
+        itemCountWatchFrame:SetScript("OnUpdate", batchUpdateCount)
+      elseif event == "BAG_UPDATE_DELAYED" then
+        -- if we *do* get a B_U_D, then cancel our fake one
+        -- item count prototype already subscribes to this event so no need to also send an internal event
+        itemCountWatchFrame:SetScript("OnUpdate", nil)
+      else
+        timer:ScheduleTimer(WeakAuras.ScanEvents, 0.2, "ITEM_COUNT_UPDATE");
+        timer:ScheduleTimer(WeakAuras.ScanEvents, 0.5, "ITEM_COUNT_UPDATE");
```

Comment:
I've tracked this down to: 0e447bb088412ac661ddbf6b32ea2509ae87363a committed in 2011 with the commit message:
"    Fixed a bug where Item Count triggers were not updating correctly with items with charges."

I suspect that's about the classic spells that cost reagents?

Reply:
Ah 2011 is cata, so maybe that's healtstone already?

Reply:
ah, yes *using* a healthstone (but not using the last charge) does indeed produce an event cascade that doesn't include a B_U_D at the end. I wonder if mana gems work the same way, one moment please.

Reply:
yeah, i get this cluster of events on using the first charge of a mana gem:
![image](https://github.com/WeakAuras/WeakAuras2/assets/24878501/3856276c-21f3-4c4a-bb90-976248194bc4)

I'll remove the old double scan and see if it breaks anything

Reply:
In my testing using a healthstone does include B_U_D, so I was about to suggest removing ScheduleTimer code here.